### ISSUE TITLE
Should I do any pre-processing for Resnet34? 

### ISSUE THREAD

**Issue by EvanAlbee** [qubvel/classification_models#25]
Hi

I have a question for you 

Should I  do any pre processing to my input data?

Is the channel order   RGB? 

Thanks !

**Comment by qubvel** [comment]
Resnet preprocessing is `identity` function, channels order is RGB

**Comment by EvanAlbee** [comment]
@qubvel  Thanks,but sorry, I have two questions:
(1)What does "Resnet preprocessing is identity function," mean? Thanks!
(2)Since I would freeze the encoder(Resnet34),I think it has to pre-process the data if the weights are trained under those operations (sub_mean,normalize....),don't we have to subtract mean or normalize the input ? 
(3) The data_format is “channels_last”, right?
I really do appreciate your reply!!!

**Comment by qubvel** [comment]
1) Identity function:
```python
def preprocess_input(x):
    return x
```
2) resnet has batchnorm layer as a first so you don't need to normalize input data 
3) yes, "channels_last"

**Comment by EvanAlbee** [comment]
@qubvel  okay but from this link:
https://github.com/qubvel/classification_models/issues/4 
I check the code the `BGRTranspose=True` for ResNet .
So the order is supposed to be BGR?

**Comment by qubvel** [comment]
No, preprocessing here: https://github.com/qubvel/classification_models/blob/ca4888745266cee2da1384259e9fd4c065dcf5bd/classification_models/resnet/models.py#L10

**Comment by yogeesh-alphaics** [comment]
hi @qubvel  .
was going thorugh the resnet34 mdoel implementation of urs , and was confused about the sequencing of batchnorm layers . as describedin the paper , batch norms are not used after each conv layer. 
will be helpful if u can explain this behaviur.
